src/animated.rs

```diff
@@ -0,0 +1,423 @@
+        if self.is_fixed() {
+            super::Transform::Fixed(self.evaluate(0.0))
+        } else {
+            super::Transform::Animated(self)
+        }
+    }
+}
+
+/// Animated ellipse.
+#[derive(Clone, Debug)]
+pub struct Ellipse {
+    /// True if the ellipse should be drawn in CCW order.
+    pub is_ccw: bool,
+    /// Position of the ellipse.
+    pub position: Value<Point>,
+    /// Size of the ellipse.
+    pub size: Value<Size>,
+}
+
+impl Ellipse {
+    pub fn is_fixed(&self) -> bool {
+        self.position.is_fixed() && self.size.is_fixed()
+    }
+
+    pub fn evaluate(&self, frame: f64) -> kurbo::Ellipse {
+        let position = self.position.evaluate(frame);
+        let size = self.size.evaluate(frame);
+        let radii = (size.width * 0.5, size.height * 0.5);
+        kurbo::Ellipse::new(position, radii, 0.0)
+    }
+}
+
+/// Animated rounded rectangle.
+#[derive(Clone, Debug)]
+pub struct Rect {
+    /// True if the rect should be drawn in CCW order.
+    pub is_ccw: bool,
+    /// Position of the rectangle.
+    pub position: Value<Point>,
+    /// Size of the rectangle.
+    pub size: Value<Size>,
+    /// Radius of the rectangle corners.
+    pub corner_radius: Value<f64>,
+}
+
+impl Rect {
+    /// Returns true if the rectangle is fixed.
+    pub fn is_fixed(&self) -> bool {
+        self.position.is_fixed() && self.size.is_fixed() && self.corner_radius.is_fixed()
+    }
+
+    /// Evaluates the rectangle at the specified frame.
+    pub fn evaluate(&self, frame: f64) -> kurbo::RoundedRect {
+        let position = self.position.evaluate(frame);
+        let size = self.size.evaluate(frame);
+        let position = Point::new(
+            position.x - size.width * 0.5,
+            position.y - size.height * 0.5,
+        );
+        let radius = self.corner_radius.evaluate(frame);
+        kurbo::RoundedRect::new(
+            position.x,
+            position.y,
+            position.x + size.width,
+            position.y + size.height,
+            radius,
+        )
+    }
+}
+
+/// Animated star or polygon.
+#[derive(Clone, Debug)]
+pub struct Star {
+    pub is_polygon: bool,
+    pub direction: f64,
+    pub position: Value<Point>,
+    pub inner_radius: Value<f64>,
+    pub inner_roundness: Value<f64>,
+    pub outer_radius: Value<f64>,
+    pub outer_roundness: Value<f64>,
+    pub rotation: Value<f64>,
+    pub points: Value<f64>,
+}
+
+// TODO: Use this.
+//impl Star {
+//    pub fn is_fixed(&self) -> bool {
+//        self.position.is_fixed()
+//            && self.inner_radius.is_fixed()
+//            && self.inner_roundness.is_fixed()
+//            && self.outer_radius.is_fixed()
+//            && self.outer_roundness.is_fixed()
+//            && self.rotation.is_fixed()
+//            && self.points.is_fixed()
+//    }
+//}
+
+/// Animated cubic spline.
+#[derive(Clone, Debug)]
+pub struct Spline {
+    /// True if the spline is closed.
+    pub is_closed: bool,
+    /// Collection of times.
```

Comment:
This is quite an unhelpful comment. I'd suggest:
```suggestion
    /// The point in time at which this should evaluate to the corresponding value in `values`.
```

Reply:
Addressed in #5.